src/go/rpk/pkg/kafka/client_franz.go

```diff
@@ -38,6 +38,7 @@ func NewFranzClient(
 		kgo.SeedBrokers(k.Brokers...),
 		kgo.ClientID("rpk"),
 		kgo.RetryTimeout(5 * time.Second),
+		kgo.MetadataMinAge(time.Second),
```

Comment:
This age is the lower bound on repeated attempts at _successful_ requests. Areas in the client can be like "hey I just failed at something, the error was pretty generic so I'm just going to back off, but maybe our metadata is out of date?" and the min age is the minimum amount of time to wait since the last metadata request before it refreshes and answers that question. There are other areas in the client that say "hey I know something isn't right, refresh right now", which bypasses this minimum window (even here though we still bound to a 10ms window to avoid pure hot loops).

_Failed_ requests do have a small bit of exponential backoff, and the the default here is to actually start at 100ms and top out at 1ms. So, if a request fails to even be issued, it'll be retried immediately.

In the context of this PR, a produce request is failing with `connect: connection refused`. I've currently coded it such that this error is in the first category (pretty generic, maybe it's metadata, maybe not), so we trigger that the metadata should be refreshed, but not immediately.

I could drop this down to 250ms, but I'd only do that for a CLI. In long running programs this is actually at 10s. I could also consider making the dial figure trigger an immediate refresh, though I'd be wary of this because in long running programs, network failures, though infrequent, are pretty common, and I wouldn't want the client going CPU crazy when everything else on a user's server may be going haywire as well.

Reply:
Dropped to 250ms :+1: